### PR TITLE
Add introspection support in OpenIdConnectConfiguration

### LINKED ISSUE #1209: OpenIdConnectConfiguration doesn't have TokenIntrospectionEndpoint
OpenIdConnectConfiguration doesn't have TokenIntrospectionEndpoint property.

It is very useful property in access token introspection scenario. 

```
var response = await client.IntrospectTokenAsync(new TokenIntrospectionRequest {
    Address = "https://demo.identityserver.io/connect/introspect", // <- here
    ClientId = "id",
    ClientSecret = "secret",
    Token = accessToken
});
```

### PR DESCRIPTION
**Pull request by kevinchalet**
Fixes https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1209.

@brentschmaltz is it something that could be included in the next release? 😃

### CODE REVIEW

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json [hunk 1]
```diff
@@ -15,6 +15,9 @@
     "id_token_encryption_alg_values_supported" : ["RSA1_5", "A256KW"],
     "id_token_encryption_enc_values_supported" : ["A128CBC-HS256","A256CBC-HS512"],
     "id_token_signing_alg_values_supported": [ "RS256" ],
+    "introspection_endpoint" : "https://login.windows.net/d062b2b0-9aca-4ff7-b32a-ba47231a4002/oauth2/introspect",
```
Comment (kevinchalet) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json:18]: Note: Azure AD doesn't support introspection so these values are fake.

**Comment by brentschmaltz** [comment]
@kevinchalet we can fit this in, thanks.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs [hunk 2]
```diff
@@ -51,6 +51,9 @@ public static class OpenIdProviderMetadataNames
         public const string IdTokenEncryptionAlgValuesSupported = "id_token_encryption_alg_values_supported";
         public const string IdTokenEncryptionEncValuesSupported = "id_token_encryption_enc_values_supported";
         public const string IdTokenSigningAlgValuesSupported = "id_token_signing_alg_values_supported";
+        public const string IntrospectionEndpoint = "introspection_endpoint";
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs:54]: @kevinchalet is there a spec we can reference for these values?

Comment (kevinchalet) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs:54]: Yep, these values are defined in the OAuth 2.0 Authorization Server Metadata specification (RFC8414): https://tools.ietf.org/html/rfc8414#section-2

Comment (brentschmaltz) [reply src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs:54]: @kevinchalet super thanks!

**Pull request merged**